spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -432,4 +405,340 @@ public void testAggregateExpressions() {
     assertThat(r1.get(10)).isEqualTo(0.25); // VAR_POP(num1)
     assertThat(r1.get(11)).isEqualTo(0.5); // VAR_SAMP(num1)
   }
+
+  @Test
+  public void testJoinQuery() {
+    writeTestDataToBigQuery();
+    Dataset<Row> df =
```

Comment:
We can extract this read logic out to a method since it is used multiple times

Reply:
Addressed